components/Board.tsx

```diff
@@ -1,17 +1,39 @@
-import React, { useEffect } from "react";
+import React, { useEffect, useState } from "react";
 
 export interface Board {
   board: number[][];
+  orientation: string;
+  isShipSelected: boolean;
+  setCurrentGridXLocation: React.Dispatch<
+    React.SetStateAction<number | undefined>
+  >;
+  setCurrentGridYLocation: React.Dispatch<
+    React.SetStateAction<number | undefined>
+  >;
 }
 
-//Creates the board for placement and for the opponent
+/** Creates the board for placement and for the opponent */
 const Board = (props: Board) => {
   return (
     <>
       {props.board.map((column, i) => (
-        <div key={i} id={`column-${i}`}>
+        <div
+          key={i}
+          id={`column-${i}`}
+          onMouseOverCapture={() => {
```

Comment:
mouseOverCapture seems more than necessary. Hovering over a `div` we don't plan to place on doesn't need to emit an event. We only need this information when the user lets go of their mouse (`onmouseup`). 

I can see that potentially messing with state updating in time for functions; Try and see if it works.

Reply:
Ah yeah, `onmouseup` should be way more convenient, but that `onMouseOverCapture` is a remnant of a function I was testing. Mainly in attempt to resolve the cursor and dragged object alignment issue. I tried to make a highlight before placement, but that code got tricky and I put that further down in the to-do list for now.

Reply:
I tried `onmouseup`, it doesn't seem to function the way I want, I'll discuss this if I have more time

Reply:
Probably have to do with `placeShip` function running first before `onmouseup` updates the state. I think it would work if the `onmouseup` just runs `placeShip` without relying on states.